src/components/settings/CertificateSection.vue

```diff
@@ -1,92 +1,89 @@
 </template>
 
 <script lang="ts">
+    import BaseSection from "@/components/common/section/BaseSection.vue";
     import { onBeforeMount, ref } from "vue";
     import { useStore } from "@/use/store/store";
-    import LoadingSpinner from "@/components/common/loading/Spinner.vue";
 
     export default {
         name: "CertificateSection",
-        components: { LoadingSpinner },
+        components: {
+            BaseSection,
+        },
         setup() {
-            const busy = ref(false);
-            const certificate = ref("");
-            let certificateBlobURL = ref("");
+            const store = useStore();
+            const isLoading = ref(true);
             const hasCertificate = ref(false);
+            const certificate = ref("");
+            const certificateDownloadURL = ref("");
 
-            onBeforeMount(async () => {
-                busy.value = true;
-                const store = useStore();
-                hasCertificate.value = await store.getters.hasCertificate;
-                if (hasCertificate.value) {
-                    await getCertificate();
-                }
-                busy.value = false;
+            onBeforeMount(() => {
+                store.getters.hasCertificate
+                    .then((alreadyHasCertificate) => (hasCertificate.value = alreadyHasCertificate))
+                    .then(() => {
+                        if (hasCertificate.value) {
+                            loadCertificate();
+                        }
+                    })
+                    .finally(() => (isLoading.value = false));
             });
 
-            async function getCertificate() {
-                busy.value = true;
-                const store = useStore();
-                certificate.value = (await store.getters.certificate()).certificate;
-                if (certificate.value == "") {
-                    hasCertificate.value = false;
-                } else {
-                    hasCertificate.value = true;
-                }
-                let blob = new Blob([certificate.value], { type: "pem" });
-                certificateBlobURL.value = URL.createObjectURL(blob);
-                busy.value = false;
+            async function loadCertificate() {
+                return store.getters
+                    .certificate()
+                    .then((certificateObject) => (certificate.value = certificateObject.certificate))
+                    .then(() => createCertificateDownloadURL(certificate.value));
+            }
+
+            function createCertificateDownloadURL(certificateContent: string) {
+                let certificateFile = new Blob([certificateContent], { type: "pem" });
+                certificateDownloadURL.value = URL.createObjectURL(certificateFile);
+            }
+
+            function createCertificate() {
+                //todo: when someone clicks and then aborts the key modal the promise won't reject/resolve resulting in indefinitely loading?!
```

Comment:
```suggestion
```
did not check this, but if the error remains, open a ticket, might be a problem in the store getter